### PR TITLE
Un-clang-format GTF_ flags

### PR DESCRIPTION
**Pull request by BruceForstall**
Remove unused `GTF_NO_OP_NO`.

Also, put some under `#ifdef` to indicate their limited use.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@briansull PTAL
cc @dotnet/jit-contrib

File: src/jit/gentree.h [hunk 1]
```diff
@@ -889,166 +885,171 @@ struct GenTree
...
-#define GTF_ICON_STR_HDL 0x60000000    // GT_CNS_INT -- constant is a string handle
-#define GTF_ICON_PSTR_HDL 0x70000000   // GT_CNS_INT -- constant is a ptr to a string handle
-#define GTF_ICON_PTR_HDL 0x80000000    // GT_CNS_INT -- constant is a ldptr handle
-#define GTF_ICON_VARG_HDL 0x90000000   // GT_CNS_INT -- constant is a var arg cookie handle
-#define GTF_ICON_PINVKI_HDL 0xA0000000 // GT_CNS_INT -- constant is a pinvoke calli handle
-#define GTF_ICON_TOKEN_HDL 0xB0000000  // GT_CNS_INT -- constant is a token handle
-#define GTF_ICON_TLS_HDL 0xC0000000    // GT_CNS_INT -- constant is a TLS ref with offset
-#define GTF_ICON_FTN_ADDR 0xD0000000   // GT_CNS_INT -- constant is a function address
-#define GTF_ICON_CIDMID_HDL 0xE0000000 // GT_CNS_INT -- constant is a class or module ID handle
-#define GTF_ICON_BBC_PTR 0xF0000000    // GT_CNS_INT -- constant is a basic block count pointer
```
Comment (briansull) [inline src/jit/gentree.h]: Thank You, I was just looking at that horrible mess today.

File: src/jit/gentree.h [hunk 2]
```diff
@@ -828,55 +836,43 @@ struct GenTree
...
 
-#define GTF_BOOLEAN 0x00040000 // value is known to be 0/1
-
-#define GTF_SMALL_OK 0x00080000 // actual small int sufficient
-
-#define GTF_UNSIGNED 0x00100000 // with GT_CAST:   the source operand is an unsigned type
-                                // with operators: the specified node is an unsigned operator
-
-#define GTF_LATE_ARG                                                                                                   \
-    0x00200000 // the specified node is evaluated to a temp in the arg list, and this temp is added to gtCallLateArgs.
```
Comment (briansull) [inline src/jit/gentree.h]: This one was the worst, splitting the define value onto the next line ugg.

**Comment by hqueue** [comment]
@dotnet-bot please test Tizen armel Cross Release Build

**Comment by BruceForstall** [comment]
@dotnet-bot test this please

**Pull request merged**